internal/app/machined/internal/phase/phase.go

```diff
@@ -103,19 +104,30 @@ func (r *Runner) runPhase(phase *Phase) error {
 		go r.runTask(task, errCh)
 	}
 
-	var result *multierror.Error
+	var (
+		result *multierror.Error
+		reboot bool
+	)
 
 	for range phase.tasks {
 		err := <-errCh
 		if err != nil {
-			log.Printf("[phase]: %s error running task: %s", phase.description, err)
+			if errors.Is(err, runtime.ErrReboot) {
+				reboot = true
+			} else {
+				log.Printf("[phase]: %s error running task: %s", phase.description, err)
+			}
 		}
 
 		result = multierror.Append(result, err)
 	}
 
 	log.Printf("[phase]: %s done, %s", phase.description, time.Since(start))
 
+	if reboot {
+		return runtime.ErrReboot
```

Comment:
this is required because what is returned by `multierror` doesn't support `errors.Is` protocol, so we can't check if error is `runtime.ErrReboot` later on in `main.go`